Source/Moya+Internal.swift

```diff
@@ -228,6 +228,18 @@ private extension MoyaProvider {
             if let dataRequest = dataRequest as? T {
                 progressAlamoRequest = dataRequest
             }
+        } else if var dataRequest = progressAlamoRequest as? DownloadRequest {
+            dataRequest = dataRequest.response(queue: queue, completionHandler: { handler in
+                let result = convertResponseToResult(handler.response, request: handler.request, data: nil, error: handler.error)
+                // Inform all plugins about the response
+                plugins.forEach { $0.didReceiveResponse(result, target: target) }
+                progressCompletion?(ProgressResponse(response: result.value))
+                completion(result)
+            })
+            
+            if let dataRequest = dataRequest as? T {
+                progressAlamoRequest = dataRequest
+            }
```

Comment:
Looks like this `else if` block looks very similar to the one above it, it's just the `progressCompletion` block that's called here and not above. Do you think there's a better abstraction we could use to avoid the repetition?